spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/InitCommand.java

```diff
@@ -148,11 +153,11 @@ private void projectGenerationOptions() {
 					.defaultsTo("maven");
 			this.format = option("format", "Format of the generated content (for example 'build' for a build file, "
 					+ "'project' for a project archive)").withRequiredArg().defaultsTo("project");
-			this.javaVersion = option(Arrays.asList("java-version", "j"), "Language level (for example '1.8')")
-					.withRequiredArg();
+			this.javaVersion = option(Arrays.asList("javaVersion", "java-version", "j"),
```

Comment:
The plan is to use kebab case consistently, not introducing a mixed case for new properties.